quic/s2n-quic/src/tests/offload_tls.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test]
+#[cfg(feature = "unstable-offload-tls")]
+fn offload_tls() {
+    use super::*;
+    use crate::provider::tls::{default, offload::Offload};
+    use s2n_quic_core::crypto::tls::testing::certificates::{CERT_PEM, KEY_PEM};
+
+    let model = Model::default();
+
+    let server_endpoint = default::Server::builder()
+        .with_certificate(CERT_PEM, KEY_PEM)
+        .unwrap()
+        .build()
+        .unwrap();
+    let client_endpoint = default::Client::builder()
+        .with_certificate(CERT_PEM)
+        .unwrap()
+        .build()
+        .unwrap();
+    let server_endpoint = Offload(server_endpoint);
+    let client_endpoint = Offload(client_endpoint);
+    test(model, |handle| {
```

Comment:
I'm surprised this is working with bach. I won't like interacting with an _actual_ async operation (interacting with a thread). My guess is it's a race condition of things actually working or not, at the very least non-deterministic.